src/NUnitFramework/framework/Constraints/CollectionConstraint.cs

```diff
@@ -35,10 +36,7 @@ protected static bool IsEmpty(IEnumerable enumerable)
             if (enumerable is ICollection collection)
                 return collection.Count == 0;
 
-            foreach (object o in enumerable)
-                return false;
-
-            return true;
+            return !enumerable.Cast<object>().Any();
```

Comment:
I wonder if `object _` would have worked to resolve the unused variable notification. This is a nice simplification, though.

It's probably not worth making more complex to squeeze out perf:
```cs
var enumerator = enumerable.GetEnumerator();
using (enumerator as IDisposable)
    return enumerator.MoveNext();
```